subprojects/composite-builds/src/integTest/groovy/org/gradle/integtests/composite/CompositeBuildArtifactTransformIntegrationTest.groovy

```diff
@@ -48,10 +57,29 @@ class CompositeBuildArtifactTransformIntegrationTest extends AbstractCompositeBu
                     artifactTransform(XForm)
                 }
             }
+            
+            task resolve {
+                def artifacts = configurations.compileClasspath.incoming.artifactView { 
+                    attributes.attribute(Attribute.of("artifactType", String), "xform")
+                }.artifacts
+                inputs.files artifacts.artifactFiles
+                doLast {
+                    artifacts.each { println it }
+                }
+            }
         """
         expect:
-        execute(buildA, "assemble")
+        execute(buildA, "resolve")
         assertTaskExecuted(":buildB", ":jar")
         assertTaskExecuted(":buildC", ":jar")
+
+        outputContains("buildB-1.0.jar (project :buildB)")
+        outputContains("buildC-1.0.jar (project :buildC)")
```

Comment:
We should probably add some context around this as well.

Reply:
Done.